### PR TITLE
Use ruby version 2.6.3

### PR DESCRIPTION
**Pull request by mjankowski**
This branch uses this PR as a base branch - https://github.com/rubygems/rubygems.org/pull/1783 - which is why the 2.5.3 commit is in here as well.

I will rebase/update this if/when the 2.5.3 PR is merged.

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Feel free to rebase this now.

**Comment by mjankowski** [comment]
Rebased.

Note that getting 2.6 branch passing on CI is contingent on rubocop upgrade PR as well - https://github.com/rubygems/rubygems.org/pull/1875

**Comment by mjankowski** [comment]
Rebased again post-rubocop-update, probably good now.

File: test/unit/dependency_test.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ class DependencyTest < ActiveSupport::TestCase
     context "that refers to a Rubygem that exists and has multiple requirements" do
       setup do
         @rubygem        = create(:rubygem)
-        @requirements   = ["< 1.0.0", ">= 0.0.0"]
+        @requirements   = [">= 0.0.0", "< 1.0.0"]
```
Comment (simi) [inline test/unit/dependency_test.rb]: Any idea why is this change needed?

Comment (mjankowski) [reply test/unit/dependency_test.rb]: The order that these get returned in changed in rubygems with the version bump, so I updated the test to match.

Comment (simi) [reply test/unit/dependency_test.rb]: Any idea what is the reson for this change? It is not good idea to just change unrelated specs on change because of fail.

Comment (mjankowski) [reply test/unit/dependency_test.rb]: In the underlying rubygems code (ie, actual rubygems, not the website), within the `Gem::Dependency` class, there is a `requirements_list` method which the `Dependency` (website app) class uses to generate the requirements list if it's not there already.

This method was changed in rubygems at some point to sort the list, and a side effect of the sorting is that this test failed.

That said, it appears as though the sorting has been backed out - https://github.com/rubygems/rubygems/commit/f2d95173d8fcbddea468db842b4671575544f1d0 - so it's possible that in a future ruby and/or rubygems release this will have to be updated again.

I don't really have a strong opinion on or too much knowledge of why it was changed in rubygems or what's more correct.

I'll push an update to keep the requirements list we send in the same, and instead change the value we expect to get out. Will also update commit message to hopefully capture the change.

Comment (simi) [reply test/unit/dependency_test.rb]: Maybe we can make the assertion not order dependent somehow so we don't to change it up and down per release.

Comment (mjankowski) [reply test/unit/dependency_test.rb]: I thought about doing that - and doing a set comparison where order doesn't matter instead ... but the thing we're testing here returned from the `#requirements` method is a string and I'm not sure what array we'd use to compare there.

The way the method is currently written, I think "return a string that puts the version reqs in whatever order rubygems put them in" is the expected behavior.

Comment (simi) [reply test/unit/dependency_test.rb]: If we don't care about the order, let's do something like:

```ruby
assert_equal @requirements.sort, @dependency.requirements.split(', ').sort
```

Comment (mjankowski) [reply test/unit/dependency_test.rb]: done in last push

Comment (simi) [reply test/unit/dependency_test.rb]: gj, looks good on CI as well

**Comment by sonalkr132** [comment]
Thanks, @mjankowski  :bowing_man: 
Sorry about the delay in getting this merged.
